lib/utils.ml

```diff
@@ -21,6 +21,14 @@ struct
     f (rev l1) l2
 
   let (@) = append
+  
+  let concat (xss : 'a list list) : 'a list =
+    let rec tail xss acc =
+      let (@) = List.rev_append in 
```

Comment:
This append changes the ordering of the elements in the list, is that intended?

Reply:
Before I was appending to the end of the intermediate list for every sub-list causing O(n^2) runtime while this time I'm treating the intermediate list as a stack using List.rev_append to push multiple items onto the stack and then reversing the stack to give the result in the correct order.
I think my binding of List.rev_append to  @ (append) is probably unhelpful and misleading.

Reply:
Ah, sorry I overlooked the reverse in the nil case, thanks for explaining :)